test/crossfilter-test.js

```diff
@@ -2114,10 +2114,18 @@ suite.addBatch({
               data.val.filter(2);
               data.remove();
               assert.deepEqual(data.val.groupSumLength.all(), [
-                { key: 6, value: 4 }
+                { key: 3, value: 3 },
```

Comment:
Just for posterity, here is the logic as I just had to spend 20 minutes working through it. Here's what I see as far as the data in the Crossfilter before the `data.remove()` line:

```
[
  {foo: 1, val: [1, 2]}, 
  {foo: 2, val: [1,2]}, 
  {foo: 3, val: [3, 4, 5]}, 
  {foo: 3, val: [1,2]},
  {foo: 1, val: [5,6,7]},
  {foo: 3, val: [6]}
]
```

Then we filter to records with `2` in the `val` array and remove those records, leaving us with:

```
[
  {foo: 3, val: [3, 4, 5]}, 
  {foo: 1, val: [5,6,7]},
  {foo: 3, val: [6]}
]
```

Giving us a total length of `val` arrays for the following keys:

3 -> 3
4 -> 3
5 -> 6
6 -> 4
7 -> 3

In this case the `groupSumLength` and `groupSumEach` groups are the same because the only filter in place is on their own dimension so these groups aren't really filtered at all.